ts_src/index.ts

```diff
@@ -174,23 +195,55 @@ export function entropyToMnemonic(
 
   const bits = entropyBits + checksumBits;
   const chunks = bits.match(/(.{1,11})/g)!;
-  const words = chunks.map(
-    (binary: string): string => {
+  const wordsAsBuffers = chunks.map(
+    (binary: string): Buffer => {
       const index = binaryToByte(binary);
-      return wordlist![index];
+      wordlist = wordlist || [];
+      return Buffer.from(wordlist[index], 'utf8');
     },
   );
 
-  return wordlist[0] === '\u3042\u3044\u3053\u304f\u3057\u3093' // Japanese wordlist
-    ? words.join('\u3000')
-    : words.join(' ');
+  const bufferSize = wordsAsBuffers.reduce(function(
+    bufferSize,
+    wordAsBuffer,
+    i,
+  ) {
+    const shouldAddSeparator = i < wordsAsBuffers.length - 1;
+    return bufferSize + wordAsBuffer.length + (shouldAddSeparator ? 1 : 0);
+  },
+  0);
+
+  const separator =
+    wordlist[0] === '\u3042\u3044\u3053\u304f\u3057\u3093' ? '\u3000' : ' ';
```

Comment:
We should preserve the comment explaining what this is (`// Japanese wordlist`)

Reply:
done here: https://github.com/MetaMask/bip39/pull/1/commits/1723d9ff5bf722d94133dfbfea5608073b36c6ba